Sources/ScriptingCore/utils.swift

```diff
@@ -0,0 +1,121 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Basics
+import Foundation
+import ScriptParse
+import TSCBasic
+
+fileprivate extension AbsolutePath {
+    var sha256Hash: String {
+        String(ByteString(stringLiteral: pathString)
+                .sha256Checksum.prefix(6))
+    }
+}
+
+/// An enumeration of the errors that can be generated by the run tool.
+public enum ScriptError: Swift.Error {
+    /// The specified file doesn't exist.
+    case fileNotFound(String)
+    /// The target path is directory
+    case isDirectory(String)
+}
+
+extension ScriptError: CustomStringConvertible {
+    public var description: String {
+        switch self {
+        case .fileNotFound(let path):
+            return "\(path) doesn't exist"
+        case .isDirectory(let path):
+            return "\(path) is a directory"
+        }
+    }
+}
+
+/// Resolves a path string to `AbsolutePath`.
+public func resolveFilePath(_ path: String) -> AbsolutePath? {
```

Comment:
I think there's a convenience initializer in `AbsolutePath` to do this.

Reply:
This is the current implementation:
```swift
public init(_ absPath: AbsolutePath, _ relStr: String) {
    self.init(absPath, RelativePath(relStr))
}
```

The second parameter is used to init a `RelativePath`, which cannot meet our need.
I'm working on a more robust implementation at [apple/swift-tools-support-core#219], but there's more to be done before it can get merged.